### PR TITLE
[Java] Implement concept exercise numbers (#1786)

### LINKED ISSUE #1783: [Java] Implement new Concept Exercise: arrays
This issue describes how to implement the `arrays` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of exercise is to learn about basic operations on arrays

## Learning objectives

- Defining an array.
- Accessing elements in an array by index.
- Updating an element in an array by index.
- Iterating over elements in an array.
- Basic array functions (like finding the index of an element in an array).

## Out of scope

- Multi-dimensional arrays.
- Memory and performance characteristics of arrays.
- Iterators.

## Concepts

- `arrays`: know how to define an array; know how to access elements in an array by index; know how to update an element in an array by index; know how to iterate over elements in an array; know of some basic functions (like finding the index of an element in an array).
- `for-loops`: know how to use a `for` loop to do iteration.

## Prerequisites

This exercise's prerequisites Concepts are:

- `classes`: know how to work with fields.
- `booleans`: know what a `boolean` is.
- `basics`: know how to work with integer numbers and how to assign and update variables.

### Hints

Nothing to report

### After

Nothing to report

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/java-representer)

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/java-analyzer)

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

### PR DESCRIPTION
**Pull request by TalesDias**
Closes #1783

### CODE REVIEW

**Comment by TalesDias** [comment]
Sorry to interrupt @mirkoperillo, but how exactly i perform the code-formating? it says i need to run Prettifier against all `.md` files (???)
Undoubtedly i didn't knew about, the `gradle check` command, so all `.java` files have style errors. but shouldn't it fail in another test?

Regards, from a simple novice in  git

**Comment by mikedamay** [comment]
@TalesDias 

To run prettier you need to install locally _node_ and then the _prettier npm_ package.  I use the _JetBrains Webstorm IDE_ to actually run _prettier_ so I am not familiar with the command line. @mirkoperillo may have more experience.

However, in the meantime, you can run _prettier_ from within the PR as follows:
- type __/format__ into an empty comment box on the PR.
- click on _Commant_
- click _Close Pull Request_
- click _Reopen Pull Request_

It will then execute an action as _bot_.

It is a little temperamental.  Give it a go and see if it works.

**Comment by mikedamay** [comment]
Regarding formatting, the code _github actions_ got involved inadvertently in response to my comment above.  For the sake of clarity, I have gone ahead and completed the formatting but the steps I gave above should be good next time you need to do this.

**Comment by TalesDias** [comment]
Hmm, right. I think i understood now.
I was not sure if i really needed to install npx,prettier and all this stuff.
I will give a chance to  Webstorm (love VScode so much)

**Comment by ynfle** [comment]
@TalesDias, you can definitely use prettier with vscode as an [extension](https://github.com/prettier/prettier-vscode).
[Here](https://www.robinwieruch.de/how-to-use-prettier-vscode/) is an article that guides you through the steps of setting it up. I haven't personally used it, but you can PM on Slack if you need some help. I know there are a whole bunch of maintainers and contributors that use it in VScode

**Comment by TalesDias** [comment]
Thank you @ynfle, i will check it out

**Comment by TalesDias** [comment]
> > While browsing the files i found ´Java/config.json´, should i fill it?
> 
> Yes. I usually do this as late as possible to avoid merge conflicts.
> 
> If you have any problems leave a comment. Remember to generate a new UUID.

How do I generate a new one? is there any CLI for this?

**Comment by mikedamay** [comment]
> How do I generate a new one? is there any CLI for this?

What OS are you on?  Google shows generators avaialble for Windows 10.  uuidgen from Terminal on Mac

If you are using an IDE then both VS Code and Intellij have GUID plugins.

**Comment by mirkoperillo** [comment]
Hi @TalesDias  sorry for the late, I was offline for my summer holidays
I'll provide you a review ASAP

Thank you for contributing to Java track

Thank you @mikedamay for the help with the review

File: languages/java/exercises/concept/numbers/.docs/after.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+One of the key aspects of working with numbers in Java is the distinction between integers (numbers with no digits after the decimal separator) and floating-point numbers (numbers with zero or more digits after the decimal separator).
+
+Java has other [datatypes][numeric-datatypes] apart from `int` and `double`
+
+```Java
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/numbers/.docs/after.md:5]: the expected label for block syntax highlight is lower-case, so `java` please fix it in all markdown files

File: languages/java/exercises/concept/numbers/.docs/after.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+int fromDouble = (int)d;
+```
+
+An `if` statement can be used to conditionally execute code. The condition of an `if` statement must be of type `boolean`. Java has no concept of _truthy_ values.
+
+```Java
+int x = 6;
+
+if (x == 5)
+{
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/numbers/.docs/after.md:69]: Normally java prefers to put the open curly bracket on the same line of the instruction
example
```
if (x == 5) {
  ....
}
```
please fix it in all markdown files

**Review by mirkoperillo — CHANGES_REQUESTED** [review]
@TalesDias good job, LGTM.
 
I ask you only a couple of minor changes and the PR will be ready to be merged.

File: languages/java/exercises/concept/numbers/.docs/after.md (hunk 2 above)
Comment (TalesDias) [reply languages/java/exercises/concept/numbers/.docs/after.md:69]: Should i use 
```
} else if (...) {
```
or 
```
} 
else if (...){
```

Comment (mirkoperillo) [reply languages/java/exercises/concept/numbers/.docs/after.md:69]: the first one

**Review by mirkoperillo — APPROVED** [review]

**Comment by mirkoperillo** [comment]
@TalesDias good work, merged!

**Pull request merged**